srml/contracts/src/lib.rs

```diff
@@ -799,10 +807,20 @@ decl_storage! {
 	}
 }
 
+/// contract uses this prefix
+pub const CHILD_CONTRACT_PREFIX: &'static [u8] = b"default:";
+
+fn prefixed_trie_id(trie_id: &[u8]) -> Vec<u8> {
+	let mut res = CHILD_CONTRACT_PREFIX.to_vec();
+	res.extend_from_slice(trie_id);
+	res
+}
+
 impl<T: Trait> OnFreeBalanceZero<T::AccountId> for Module<T> {
 	fn on_free_balance_zero(who: &T::AccountId) {
 		if let Some(ContractInfo::Alive(info)) = <ContractInfoOf<T>>::get(who) {
-			child::kill_storage(&info.trie_id);
+			child::child_trie(&prefixed_trie_id(&info.trie_id[..])[..])
```

Comment:
This access pattern is repeated a few times. I think it's worth defining

```rust
pub(crate) fn lookup_child_trie(id: TrieId) -> Option<ChildTrie> {
    child::child_trie(&prefixed_trie_id(&trie_id)[..]))
}
```

(you could also inline the definition of `prefixed_trie_id` in that function)